build_wheels.py

```diff
@@ -172,28 +177,239 @@ def assemble_requirements(idf_branches: List[str], idf_constraints: List[str]) -
+                    print(Fore.GREEN + f'++ {new_requirement}')
+
+
+        if len(not_in_exclude) == len(exclude_list):
+            if print_requirements:
+                print(Style.RESET_ALL + str(requirement))
+            new_assembled_requirements.add(requirement)
+
+        not_in_exclude.clear()
+
+    if print_requirements:
+        print(Fore.BLUE + '---------- END OF REQUIREMENTS ----------' + Style.RESET_ALL)
+
+    return new_assembled_requirements
+
+
+# --- Build wheels ---
+failed_wheels = 0
+succeeded_wheels = 0
+
+def build_wheels(requirements: set):
+    """Build Python wheels"""
+    global failed_wheels, succeeded_wheels
+
+    dir = f'{os.path.curdir}{(os.sep)}downloaded_wheels'
+    for requirement in requirements:
+        # --only-binary requirement wheel build
+        if non_classic_requirement and non_classic_requirement[0].replace('--only-binary ', '') in requirement.name:
+            only_bin = non_classic_requirement[0].replace('--only-binary', '').strip()
+            out = subprocess.run(
+                ['python', '-m', 'pip', 'wheel', f'{requirement}',
+                 '--find-links', f'{dir}', '--wheel-dir', f'{dir}', '--only-binary', f'{only_bin}'],
+                 stdout=subprocess.PIPE, stderr=subprocess.PIPE
+                )
+            print(out.stdout, out.stderr)
+            non_classic_requirement.remove(non_classic_requirement[0])
+            continue
+
+        # requirement wheel build
+        out = subprocess.run(
+            ['python', '-m', 'pip', 'wheel', f'{requirement}',
+             '--find-links', f'{dir}', '--wheel-dir', f'{dir}'],
+             stdout=subprocess.PIPE, stderr=subprocess.PIPE
+             )
+
+        print(out.stdout.decode('utf-8'))
+        if out.stderr:
+            print(Fore.RED + out.stderr.decode('utf-8') + Style.RESET_ALL)
+
+        if out.returncode != 0:
+            failed_wheels += 1
+        else:
+            succeeded_wheels += 1
 
 
 def main():
-    """Builds Python wheels for IDF dependencies"""
+    """Builds Python wheels for ESP-IDF dependencies for master and release branches
+    grater or equal to specified"""
+
     idf_repo_branches = fetch_idf_branches()
     idf_branches = get_used_idf_branches(idf_repo_branches)
     print(f'ESP-IDF branches to be downloaded requirements for:\n{idf_branches}\n')
 
     idf_constraints = get_constraints_versions(idf_branches)
     print(f'ESP-IDF constrains files versions to be downloaded requirements for:\n{idf_constraints}\n')
 
-    requirements_set = assemble_requirements(idf_branches, idf_constraints)
+    requirements = assemble_requirements(idf_branches, idf_constraints)
+
+    exclude_list = yaml_to_requirement('exclude_list.yaml', exclude=True)
+    excluded_requirements = exclude_from_requirements(requirements, exclude_list)
+
+    include_list = yaml_to_requirement('include_list.yaml')
+    print(Fore.BLUE + '---------- ADDITIONAL REQUIREMENTS ----------' + Style.RESET_ALL)
```

Comment:
(optional) We can avoid all of these string concatenations:

```python
print(str1, str2, str3)
```

Also, probably we could add wrappers like `print_blue()`, or similar.

Reply:
Also please check all prints. Several colors are used but not for the warning in line 43.

Reply:
It might be worth introducing a helper function if it is frequently used...

See in this project: 
- https://github.com/espressif/conventional-precommit-linter/blob/master/conventional_precommit_linter/helpers.py#L8-L9
- https://github.com/espressif/conventional-precommit-linter/blob/master/conventional_precommit_linter/hook.py#L160